File: src/main/java/com/medicalsystem/service/impl/PatientServiceImpl.java
package com.medicalsystem.service.impl;

import com.medicalsystem.model.Form;
import com.medicalsystem.model.Patient;
import com.medicalsystem.repository.PatientRepository;
import com.medicalsystem.service.PatientService;
import lombok.AllArgsConstructor;
import lombok.extern.java.Log;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@Transactional
@AllArgsConstructor(onConstructor = @__(@Autowired))
@Log
public class PatientServiceImpl implements PatientService {

    private final PatientRepository patientRepository;

    @Override
    public List<Patient> getAll() {
        return patientRepository.findAll();
    }

    @Override
    public List<Patient> getAllByForm(Form form) {
        return patientRepository.findAllByForm(form);
    }

    @Override
    public Patient getById(Long id) {
        return patientRepository.findOne(id);
    }

    @Override
    public boolean existsById(Long id) {
        return patientRepository.exists(id);
    }

    @Override
    public Patient save(Patient patient) {
        return patientRepository.save(patient);
    }

    @Override
    public void deleteById(Long id) {
        patientRepository.delete(id);
    }

    @Override
    public boolean create(long patientId) {
        if (existsById(patientId)) {
            log.severe("Patient exists with ID: " + patientId);
            return false;
        }
        Patient patient = new Patient();
        patient.setId(patientId);
        save(patient);
        log.info("Patient created: " + patientId);
        return true;
    }
}

File: src/main/kotlin/com/medicalsystem/service/impl/PatientServiceImpl.kt
package com.medicalsystem.service.impl

import com.medicalsystem.factory.FieldValueFactory
import com.medicalsystem.model.Form
import com.medicalsystem.model.Patient
import com.medicalsystem.model.value.FieldValue
import com.medicalsystem.repository.FormRepository
import com.medicalsystem.repository.PatientRepository
import com.medicalsystem.service.FieldService
import com.medicalsystem.service.FieldValueService
import com.medicalsystem.service.FormService
import com.medicalsystem.service.PatientService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional
class PatientServiceImpl @Autowired constructor(
        val patientRepository: PatientRepository,
        val formRepository: FormRepository,
        val fieldService: FieldService,
        val fieldValueService: FieldValueService) : PatientService(patientRepository) {

    override fun create(id: String, form: Form): Patient? =
            if (exists(id)) {
                null
            } else {
                save(Patient(id, form))
            }

    override fun create(patientId: String, formName: String): Boolean {
        val form: Form? = formRepository.findByName(formName)
        return if (exists(patientId) || form == null) {
            false
        } else {
            val patient = Patient(patientId, form)
            patientRepository.saveAndFlush(patient)
            fieldService.getAllByForm(form).forEach {
                val fieldValue: FieldValue<*> = FieldValueFactory.fromFieldAndStringValue(it, "")
                fieldValue.patient = patient
                fieldValueService.save(fieldValue)
            }
            true
        }
    }

    override fun getAllByForm(form: Form): List<Patient> =
            patientRepository.findAllByForm(form)
}